core/src/repair_service.rs

```diff
@@ -518,20 +527,41 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        delay_threshold: Duration,
     ) -> Vec<ShredRepairType> {
         if max_repairs == 0 || slot_meta.is_full() {
             vec![]
         } else if slot_meta.consumed == slot_meta.received {
+            // check delay time of last shred
+            if slot_meta.received > 0 {
+                if let Some(shred_data) = blockstore
+                    .get_data_shred(slot, slot_meta.received - 1)
+                    .unwrap()
+                {
+                    let reference_tick =
+                        u64::from(shred::layout::get_reference_tick(&shred_data).unwrap());
+                    let ticks_since_first_insert = DEFAULT_TICKS_PER_SECOND
+                        * (timestamp() - slot_meta.first_shred_timestamp)
+                        / 1000;
+                    if ticks_since_first_insert
+                        < reference_tick + delay_threshold.as_millis() as u64 / MS_PER_TICK
```

Comment:
One downside of moving this outside of blockstore - I'm assuming we're recomputing milliseconds --> ticks every time now

Reply:
I could change the parameter to use ticks rather than Duration to avoid the calculation if you think it's preferable.

Reply:
I think I prefer ticks, but I'm not going to lose sleep over it :)

Reply:
ticks now passed as parameter